### PR TITLE
make all change in selinux mode persistent

### PR DESCRIPTION
**Pull request by car-da**
I prefer that all selinux mod changes be persistent so they survive the reboot of the server.
Please check if you are ok with that.

### CODE REVIEW

File: linux/system/selinux.sls [hunk 1]
```diff
@@ -10,21 +10,29 @@ include:
 
 selinux_config:
   cmd.run:
-  - names:
-    - "sed -i 's/enforcing/disabled/g' /etc/selinux/config; setenforce 0"
-    - "sed -i 's/permissive/disabled/g' /etc/selinux/config; setenforce 0"
-  - unless: cat '/etc/selinux/config' | grep 'SELINUX=disabled'
+  - name: "sed -i 's/SELINUX=[a-z][a-z]*$/SELINUX={{ system.selinux }}/' /etc/selinux/config"
+  - unless: cat '/etc/selinux/config' | grep 'SELINUX={{ system.selinux }}'
```
Comment (fpytloun) [inline linux/system/selinux.sls]: Useless use of cat, it's better to use this:
```
grep 'SELINUX={{ system.selinux }}' /etc/selinux/config
```

File: linux/system/selinux.sls [hunk 2]
```diff
@@ -10,21 +10,29 @@ include:
...
+  selinux.mode
 
 {%- else %}
 
 selinux_config:
-  selinux.mode:
-  - name: {{ system.get('selinux', 'permissive') }}
+  cmd.run:
+  - name: "sed -i 's/SELINUX=[a-z][a-z]*$/SELINUX={{ system.selinux }}/' /etc/selinux/config"
+  - unless: cat '/etc/selinux/config' | grep 'SELINUX={{ system.selinux }}'
```
Comment (fpytloun) [inline linux/system/selinux.sls]: same here

Comment (fpytloun) [inline linux/system/selinux.sls]: I think you could also reduce the code as it's duplicated for disabled vs. anything else action. Use the same states but decide on mode eg.:
```
{%- if system.selinux == 'disabled' %}
  {%- set mode = 'permisive' %}
{%- else %}
  {%- set mode = system.selinux %}
{%- endif %}
```

**Comment by car-da** [comment]
Code was reworked to correspond with comments.

File: linux/system/selinux.sls [hunk 3]
```diff
@@ -7,30 +7,20 @@ include:
...
-  - unless: cat '/etc/selinux/config' | grep 'SELINUX={{ system.selinux }}'
-  - require:
-    - pkg: linux_repo_prereq_pkgs
-
-permisive:
-  selinux.mode
-
+	{%- set mode = 'permissive' %}
 {%- else %}
+	{%- set mode = {{ system.selinux }} %}
```
Comment (fpytloun) [inline linux/system/selinux.sls]: You should use 2 spaces instead of tabs to keep unified indentation

File: linux/system/selinux.sls [hunk 4]
```diff
@@ -1,30 +1,28 @@
 {%- from "linux/map.jinja" import system with context %}
-{%- if system.enabled %}
+{%- if system.selinux is defined %}
```
Comment (fpytloun) [inline linux/system/selinux.sls]: IMO you should keep if `system.enabled` here or extend it to `if system.enabled and system.selinux is defined`.

**Comment by fpytloun** [comment]
Thank you :)

**Pull request merged**